### ISSUE TITLE
About example: workthrough-tensorflow-object-detection

### ISSUE THREAD

**Issue by bobleegogogo** [developmentseed/label-maker#67]
Hello there,
          I make this new issue because I got some question about the train and test samples split steps in this example. For the demo area of Mexico, the area of interest consist of 227 tiles in total, so if my understanding is right that, this total 227 tiles should be split into 181 training samples and 46 test samples (using 0,8 split index). But it seem the output using tf_records_generation.py from the example is not like this.
           I have a look on the code for tf_records_generation.py:
```
for tile in tiles:
        bboxes = labels[tile].tolist()
        width = 256
        height = 256
        if bboxes:
            for bbox in bboxes:
                if bbox[4] == 1:
                    cl_str = "building"
                    bbox = [max(0, min(255, x)) for x in bbox[0:4]]
                    y = ["{}.jpg".format(tile), width, height, cl_str, bbox[0], bbox[1], bbox[2], bbox[3]]
                    tf_tiles_info.append(y)

    split_index = int(len(tf_tiles_info) * 0.8)
    column_name = ['filename', 'width', 'height', 'class', 'xmin', 'ymin', 'xmax', 'ymax']
    df = pd.DataFrame(tf_tiles_info, columns=column_name)
    # shuffle the dataframe
    df = df.sample(frac=1)
    train_df = df[:split_index]
    test_df = df[split_index:]
``` 
split_index = int(len(tf_tiles_info) * 0.8), this split here is split by building features number, right ? From my point of view, we have to split the total tiles and then as well as the building features in those tiles. Otherwise, the test tiles can not be promising to be independent of the training tiles, and the results of test make no sense anymore. If I did some mistakes here or I understand wrong, please feel free to point out.

Best; 

Hao

**Comment by bobleegogogo** [comment]
I try to modify this part a lot bit by:


    split_index= int(len(tiles) * 0.8)

    tiles_train=tiles[:split_index]
    tiles_validation=tiles[split_index:]

    tf_tiles_info_train = []
    tf_tiles_info_validation = []

    for tile_train in tiles_train:
        bboxes = labels[tile_train].tolist()
        width = 256
        height = 256
        if bboxes:
            for bbox in bboxes:
                if bbox[4] == 1:
                    cl_str = "building"
                    bbox = [max(0, min(255, x)) for x in bbox[0:4]]
                    y = ["{}.jpg".format(tile_train), width, height, cl_str, bbox[0], bbox[1], bbox[2], bbox[3]]
                    tf_tiles_info_train.append(y)

    for tile_validation in tiles_validation:
        bboxes = labels[tile_validation].tolist()
        width = 256
        height = 256
        if bboxes:
            for bbox in bboxes:
                if bbox[4] == 1:
                    cl_str = "building"
                    bbox = [max(0, min(255, x)) for x in bbox[0:4]]
                    z = ["{}.jpg".format(tile_validation), width, height, cl_str, bbox[0], bbox[1], bbox[2], bbox[3]]
                    tf_tiles_info_validation.append(z)

    
    column_name = ['filename', 'width', 'height', 'class', 'xmin', 'ymin', 'xmax', 'ymax']
    df_train = pd.DataFrame(tf_tiles_info_train, columns=column_name)
    df_validation = pd.DataFrame(tf_tiles_info_validation, columns=column_name)
    # shuffle the dataframe
    df_train = df_train.sample(frac=1)
    df_validation = df_validation.sample(frac=1)
    train_df = df_train
    test_df = df_validation
    print("You have {} training tiles and {} test tiles ready".format(
        len(set(train_df['filename'])), len(set(test_df['filename']))))

    tiles_dir = op.join(os.getcwd(), 'tiles')
    train_dir = op.join(os.getcwd(), 'images', 'train')
    test_dir = op.join(os.getcwd(), 'images', 'test')

**Comment by Geoyi** [comment]
@bobleegogogo, thanks for writing this up. 
For a small training dataset is better to slip up by the building features instead of by tiles. Otherwise, you can end up with having more buildings in validation set but fewer buildings go to train the models. But if you have hug dataset for object detection, it totally makes more sense to slip up the train and validation by tiles.

**Comment by bobleegogogo** [comment]
Hello @Geoyi, I could understand that in case to avoid to have few training sample and split by building features, but problem would come when validating the model. Since in the worktrough, all the feature in the test tiles are using for validation, but notice here: some of the features have been using for training, so if we still consider them in validating step, then the result of validation is not reliable anymore. Isn’t it?

**Comment by Geoyi** [comment]
yay, that will slightly definitely overestimate the model result, @bobleegogogo. You should definitely split the dataset by tiles if you use this model on a large dataset. 
By the way, the updated IOU is this one, and I forgot to merge back to the master branch, and will definitely merge it back to the main branch later: https://github.com/developmentseed/label-maker/commit/eeede28a134451e91a9c978053b9eb85f8ffea53

**Comment by bobleegogogo** [comment]
@Geoyi , Yes, I think the problem is clear now, thanks fo your time!
